globus_sdk/response.py

```diff
@@ -15,13 +15,16 @@ class GlobusResponse(object):
     ``TypeError``.
 
     >>> print("Response ID": r["id"]) # alias for r.data["id"]
+
+    ``GlobusResponse`` objects *always* wrap some kind of data to
+    return to a caller. Most basic actions on a GlobusResponse are
+    just ways of interacting with this data.
     """
-    def __init__(self, data):
-        """
-        ``GlobusResponse`` objects *always* wrap some kind of data to
-        return to a caller. Most basic actions on a GlobusResponse are
-        just ways of interacting with this data.
-        """
+    def __init__(self, client, data):
```

Comment:
I think this is a breaking change unless we're going to try and assert that `GlobusResponse` is somehow private.

Reply:
I thought about this -- nobody other than us is probably using this initializer, but it *is* a backwards incompatible way to add the correct.
Would you be satisfied by making`client` an optional argument?

Reply:
I had the same thought as @jaswilli when I originally saw this; just structuring this as `self, data, client=None` would avoid any possibility of drama and I think that you wouldn't have needed so many changes in the test suite.